### ISSUE TITLE
Configurable Schematics

### ISSUE THREAD

**Issue by Mekomancer** [Anuken/Mindustry-Suggestions#3636]
### Describe the content or mechanics you are proposing.

Schematics could have different configurations. for example, a Schematic for a stand-alone Duo could have 2 configurations, the first could be a copper configuration, for copper deposits, and the second one could be a coal configuration, for coal deposits.
Ideally the configurations could be automatically chosen by conditions that can be edited by the player. There would also a key for changing the configuration manually, such as TAB or SHIFT.

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

It would allow for less schematics while also quicker and less repetitive building,

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by Bassem-Const** [comment]
As in like 2 schematics in the same slot?
Otherwise that's not allowed based on the readme, I don't really know.

**Comment by Mekomancer** [comment]
Yes, like 2 schematics in the same slot, but I was thinking that more than just 2 might be helpful.

**Comment by Mekomancer** [comment]
Another thing that would make this so much harder and complex, so it may not be that good, would be to have multiple configuration "dimesons", using the duo analogy, have a second configuration "dimeson" that chooses the wall type, copper or titanium, so you would pick coal or copper, and then you would pick normal or large walls. although this would only work if there are no conflicts, or it is a simple replacement, like the wall type.
